fx.go

```diff
@@ -19,3 +19,149 @@
 // THE SOFTWARE.
 
 package fx // import "go.uber.org/fx"
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
+	Stop()
+}
+
+// Service controls the lifecycle of an fx service
+type Service struct {
+	g              *dig.Graph
+	modules        []Module
+	components     []Component
+	scopeCloser    io.Closer
+	logger         *zap.Logger
+	loggerCloserFn func()
```

Comment:
zap logger returns this as a closer. Right now it unsets the globals we set. This is not strictly required since at service stop, we don't care about this anymore. But it's just good practice to call the closer fn. It can be extended in future for instance and we will be covering our bases this way.